com/win32com/test/testmakepy.py

```diff
@@ -44,9 +47,43 @@ def TestBuildAll(verbose=1):
     return num
 
 
+def _TestEnsureModule(info):
+    # This used to fail when called concurrently from multiple processes. See issue 1923.
+    # The issue only happens when bForDemand is set as that creates a package instead of
+    # a single module.
+    tinfo = (info.clsid, info.lcid, int(info.major), int(info.minor))
+    mod = gencache.EnsureModule(*tinfo, bForDemand=1)
+    if makepy.bForDemandDefault:
+        for name in mod.NamesToIIDMap:
+            makepy.GenerateChildFromTypeLibSpec(name, tinfo)
+
+
+def TestBuildConcurrent(verbose=1):
+    # Pick any type library
+    info = next(iter(selecttlb.EnumTlbs()))
+
+    if verbose:
+        print(f"{info.desc} ({info.dll})")
+
+    # Call EnsureModule from multiple processes concurrently.
+    nprocs = 16
+    with multiprocessing.Pool(nprocs) as p:
+        p.map(_TestEnsureModule, [info for i in range(nprocs)])
+
+    return nprocs
+
+
 def TestAll(verbose=0):
+    gen_path = gencache.GetGeneratePath()
+    if os.path.exists(gen_path) and os.path.isdir(gen_path):
```

Comment:
`os.path.isdir` already return `False` if location doesn't exist.

```suggestion
    if os.path.isdir(gen_path):
```